hook-worker/src/worker.rs

```diff
@@ -115,39 +115,47 @@ impl<'p> WebhookWorker<'p> {
     }
 
     /// Wait until a job becomes available in our queue.
-    async fn wait_for_job<'a>(
-        &self,
-    ) -> Result<PgJob<WebhookJobParameters, WebhookJobMetadata>, WorkerError> {
+    async fn wait_for_job<'a>(&self) -> PgJob<WebhookJobParameters, WebhookJobMetadata> {
         let mut interval = tokio::time::interval(self.poll_interval);
 
         loop {
             interval.tick().await;
             self.liveness.report_healthy().await;
 
-            if let Some(job) = self.queue.dequeue(&self.name).await? {
-                return Ok(job);
+            match self.queue.dequeue(&self.name).await {
+                Ok(Some(job)) => return job,
+                Ok(None) => continue,
+                Err(error) => {
+                    error!("error while trying to dequeue job: {}", error);
+                    continue;
```

Comment:
Maybe we could have an extra `interval.tick` or similar here to slow down if we encounter an error... But maybe for later

Reply:
We could also just do a naive sleep in this branch. We're polling at 100ms currently and I could just sleep for a second to add some backoff?

Reply:
Eh, I'll see how it goes. For now, we have the 100ms (as currently configured) interval, which prevents us from being in a super hot retry loop.